app/frontend/Shared/ContextMenu.ts

```diff
@@ -273,13 +270,23 @@ export async function openBrowser(url: URLString) {
   await appGlobal.remoteApp.openExternalURL(url);
 }
 
-export function download(url: URLString, window: any, howSaveAsDialog: boolean, filename?: string) {
-  let a = document.createElement("a");
-  a.href = url;
-  a.setAttribute("target", "_blank"); // for non-same-origin images
-  if (filename) {
-    a.setAttribute("download", filename); // TODO sanitize filename
+export async function download(url: URLString, win: any, howSaveAsDialog: boolean, filename?: string) {
+  let urlObj = new URL(url);
+  let blob: Blob;
+  if (urlObj.protocol == "http:" || urlObj.protocol == "https:") {
+    let res = await fetch(urlObj.href);
+    filename = urlObj.pathname.split("/").pop();
+    blob = await res.blob();
+  } else if (urlObj.protocol == "data:") {
+    blob = await dataURLToBlob(url);
+  } else {
```

Comment:
While you're at it, can you please also handle `blob:` URLs? We have them, too. They should obviously be trivial in this case, given that you need a `Blob` object.